### PR TITLE
RSC can create a new job

### PR DESCRIPTION
**Pull request by Prozak8**
# PT
https://www.pivotaltracker.com/story/show/161589630

-Added feature test with sad path
- Added many columns for jobs model
- Added new, create action
- Create form for adding job information
- Rendered page using js in order to show forms in modals/pop-up
screen shot 2018-11-05 at 13 54 45

### CODE REVIEW

**Review by angiecstn — APPROVED** [review]

**Comment by rupertlion** [comment]
Sorry to come at this at this late hour, but if I read this right, we only have a name field when we create the job in this PR. There are about 20 fields for the job (of which around 10 are mandatory) - according to the spreadsheet that we've been shared. See also https://projects.invisionapp.com/freehand/document/UBTPWsOFO   Perhaps we add these at another time but I suspect worth a conversation tomorrow.

**Review by RhonGabriel — APPROVED** [review]

File: app/views/users/new.js.erb [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+    let errorMessageDisplayElement = document.getElementById('error_messages')
+    let message = event.detail[0].message
+    errorMessageDisplayElement.innerHTML = message
+})
+
+form.addEventListener('ajax:success', (event) => {
+    let errorMessageDisplayElement = document.getElementById('messages')
+    let message = event.detail[0].message
+    let model = document.getElementById('modal')
+    errorMessageDisplayElement.innerHTML = message
```
Comment (Noelryn) [inline app/views/users/new.js.erb:19]: Need to change this name from errorMessageDisplayElement to messageDisplayElement on line 16 and 19.

File: features/rsc_can_add_new_job.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    As a RSC,
+    In order to create jobs,
+    I would like to click on the “+” button and fill in the jobs information.
+
+    Background: Signed in as RSC on dashboard
+        Given I am on the landing page
+        And I click on "Add New Job"
+        Then I should see "Create New Job"
+
+    Scenario: RSC creates a new job
```
Comment (Noelryn) [inline features/rsc_can_add_new_job.feature:11]: Can this be one scenario instead of two

File: features/rsc_can_add_new_job.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        And I click on "Submit"
+        Then I should see "The job was successfully created"
+        And I should see "Heart Surgeon"
+
+    Scenario: RSC fills out form incorrectly
+        And I fill in "Name" with " "
+        And I click on "Submit"
+        Then I should see "Name can't be blank"
+
+    Scenario: RSC wants to add new job but decides not to
```
Comment (Noelryn) [inline features/rsc_can_add_new_job.feature:23]: And can't this be one sad path scenario

**Review by Noelryn — CHANGES_REQUESTED** [review]

File: app/views/users/new.js.erb (hunk 1 above)
Comment (RhonGabriel) [reply app/views/users/new.js.erb:19]: Updated. We decided to cancel the (ajax) event listener for displaying success messages, as we are already doing it through the controller.

**Review by angiecstn — APPROVED** [review]

**Review by Noelryn — APPROVED** [review]

File: spec/models/job_spec.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 RSpec.describe Job, type: :model do
   describe 'DB table' do
```
Comment (oliverochman) [inline spec/models/job_spec.rb:2]: You guys dont test for all of the columns in the job model, why?

File: app/controllers/jobs_controller.rb [hunk 5]
```diff
@@ -1,5 +1,25 @@
+# frozen_string_literal: true
+
 class JobsController < ApplicationController
   def index
     @jobs = Job.all
   end
+
+  def new
+    @job = Job.new
```
Comment (oliverochman) [inline app/controllers/jobs_controller.rb]: why do you need this to be an instance variable?

**Pull request merged**